pkg/apis/hive/v1alpha1/syncset_types.go

```diff
@@ -65,6 +65,23 @@ type SyncObjectPatch struct {
 	PatchType string `json:"patchType,omitempty"`
 }
 
+// SecretResource represents an existing secret object to be synced
+type SecretResource struct {
+	Source corev1.ObjectReference `json:"source"`
```

Comment:
Enforcing that this is a secret will be the job of the controller and result in an error sent to the associated `SyncSetInstance` status.

Reply:
Is the webhook validation not sophisticated enough to handle the enforcement? Or are you arguing that it is more appropriate for the controller to handle it?

Reply:
I'm not sure if the webhook validation is or isn't but I can look into it.

Reply:
The webhook validation is "hiveadmission" that Thomas put together. Benefit of using it would be that the caller immediately gets their API request rejected, rather than the API server accepting it and the error only showing up in condition if you went to look for it. Admission would probably be the best spot for this one, it cannot do any etcd loading/querying, but in this case we can check the type easily and reject if not a secret.